CMakeLists.txt

```diff
@@ -36,9 +36,15 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/" ${CMAKE_MODUL
 include(FindPkgConfig)
 
 # ---[ Release/Debug specific flags
-if(CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
-  add_definitions("-DBOOST_DISABLE_ASSERTS -DEIGEN_NO_DEBUG")
+if(MSVC)
+  # add_definitions doesn't support generator expressions - see https://gitlab.kitware.com/cmake/cmake/issues/15374
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DBOOST_DISABLE_ASSERTS -DEIGEN_NO_DEBUG")
```

Comment:
Will the following (with no conditions) work on all platforms? It might be simpler and more straightforward:
```
set(CMAKE_CXX_FLAGS_RELEASE ...
set(CMAKE_CXX_FLAGS_RELWITHDEBINFO ...
```